### PR TITLE
Query params in search url

### PR DESCRIPTION
**Pull request by Jake-Lindvall**
NOTE: lower priority than current MRs out there. Definitely pull this down and test multiple searches in a row, or editing the url and hitting enter. Should be able to navigate back and forth between multiple searches if all goes well, too :)

## Primary
* New action creator and type for when the search results page loads up
* Cleaned up how the url with params is handled and pushed into history in the serach component
* Now using the new `useQuery()` custom hook for the first time ever! See it in action combined with `useEffect()` in the search results component
* Changed the in-code routing order in the `<Switch>` regarding `/search`; because of how params are handled, we can't rely on the `exact` prop

## Secondary
* Cleaned up and removed some unnecessary things, particularly with props we aren't using from some container components
* Fixed some apparent errors in the page wrapper styles - logic seemed inverted for light vs dark mode
* Now using type constants - we appeared to accidentally be using literal strings still in the reducers
* Added a missing type that we now need by using imported type constants in the reducer (`HANDLE_SEARCH_CLOSE`; this may change in upcoming search modal updates)
* Grammar and spelling in some places

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Overall, the code itself looks good to me! Great use of `useQuery`.

Just some merge conflicts with the newer pushes to dev.

File: src/Actions/searchResultActions.js
Comment (SelcukEdma) [inline src/Actions/searchResultActions.js:3]: Just to Question : If possible, can we pass `params` to `searchResultRequest` action, that way we dont have create this action. It seems they are doing the same.

**Comment by SelcukEdma** [comment]
Overall is good & clean. 👍

File: src/Actions/searchResultActions.js
Comment (ghost) [reply src/Actions/searchResultActions.js:3]: Yeah I remember discussing this with him as well... now I am seeing how these are redundant. What do you think? @Jake-Lindvall

Comment (Jake-Lindvall) [reply src/Actions/searchResultActions.js:3]: Did you mean pass it to `searchHandleInput` action? As it is right now, the logic in `SearchResults` looks like:

```
if (query) {
  searchResultPageLoad(query);
  searchResultRequest();
}
```

For a little clarity, here's why I have it as a separate action, though:
* `searchHandleInput` seems to depend on `e.target.value`, and I didn't want to refactor everything around that
* `searchResultPageLoad` makes it simpler and runs every page load, looking to update the state if possible and avoids being concerned at all with pretending there was a user input
* Having a separate action and action type is clearer for debugging and other semantic purposes - it isn't an input, it really is just a page load

If that still seems silly, I totally understand and can update it 👍

File: src/Components/Scenes/SearchResults/SearchResults/SearchResults.js
Comment (robertwollny-pnmac) [inline src/Components/Scenes/SearchResults/SearchResults/SearchResults.js:5]: Love the use of a custom hook.

File: src/Actions/searchResultActions.js
Comment (robertwollny-pnmac) [reply src/Actions/searchResultActions.js:3]: I still did not see the drop-down for users when you go to the Request Assets part of the App. Other than that all the logic and routing look great to me. There might be a little refactoring due to how we now handle the search icon now but once that gets revised I will approve.

Comment (ghost) [reply src/Actions/searchResultActions.js:3]: @robertwollny-pnmac it's due to the merge conflicts on his remote branch. Just confirmed with @Jake-Lindvall that the dropdown works on his local branch with the merge conflicts resolved 👍

Comment (Jake-Lindvall) [reply src/Actions/searchResultActions.js:3]: Pushing up changes; test it out and approve if it looks good

@robertwollny-pnmac Agreed on the refactoring - things behave slightly different now, because clicking the menu item for search directs right to `/search`, but it shouldn't be too much future work :)

**Review by ghost — APPROVED** [review]

**Review by robertwollny-pnmac — APPROVED** [review]
Looks good 🔥

**Pull request merged**
